src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -0,0 +1,1885 @@
+				}
+
+				return true;
+			}
+
+			string = iterator.next();
+		} while (!responseString.endsWith(string));
+
+		if (this.logger.isTraceEnabled()) {
+			this.logger.trace("Done reading, found error string: " + string + " from: " + this.getHost() + " port: " + this.getPort());
+		}
+
+		// if it has ACK packet first -> Error packet, 1 Completion packet left to receive from Device
+		if (hasACK) {
+			super.read(command);
+		}
+
+		throw new CommandFailureException(this.getHost(), commandString, responseString);
+	}
+
+	/**
+	 * This method is used to digest the response received from the device
+	 *
+	 * @param response This is the response to be digested
+	 * @param sequenceNum This is the sequence number of send packet
+	 * @param commandType This is the type of command to be digested
+	 * @param expectedCommand This is the expected command to be digested
+	 * @return Object This returns the result digested from the response.
+	 */
+	public Object digestResponse(byte[] response, int sequenceNum, CommandType commandType, Command expectedCommand) {
+
+		if (response[1] == PayloadType.REPLY.getCode()) {
+			byte[] responseSeqNum = Arrays.copyOfRange(response, 4, 8);
+			byte[] expectedSeqNum = convertIntToByteArray(sequenceNum);
+
+			if (Arrays.equals(expectedSeqNum, responseSeqNum)) {
+				int payloadLength = response[3];
+				byte[] reply = Arrays.copyOfRange(response, 8, 8 + payloadLength);
+				byte currentValue = reply[2];
+
+				if (commandType == CommandType.COMMAND) {
+					if (!Arrays.equals(ReplyStatus.COMPLETION.getCode(), reply)) {
+						if (this.logger.isErrorEnabled()) {
+							this.logger.error("error: Unexpected completion packet: " + this.host + " port: " + this.port);
+						}
+						throw new IllegalStateException("Unexpected completion packet");
+					}
+				} else if (commandType == CommandType.INQUIRY) {
+					switch (expectedCommand) {
+						case POWER: {
+							Optional<PowerStatus> powerStatus = Arrays.stream(PowerStatus.values())
+									.filter(status -> status.getCode() == currentValue)
+									.findFirst();
+
+							return powerStatus.orElse(null);
+						}
+						case FOCUS_MODE: {
+							Optional<FocusMode> focusMode = Arrays.stream(FocusMode.values())
+									.filter(mode -> mode.getCode() == currentValue)
+									.findFirst();
+
+							return focusMode.orElse(null);
+						}
+						case AE_MODE: {
+							Optional<AEMode> aeMode = Arrays.stream(AEMode.values())
+									.filter(mode -> mode.getCode() == currentValue)
+									.findFirst();
+							return aeMode.orElse(null);
+						}
+						case AUTO_SLOW_SHUTTER: {
+							Optional<SlowShutterStatus> slowShutterStatus = Arrays.stream(SlowShutterStatus.values())
+									.filter(mode -> mode.getCode() == currentValue)
+									.findFirst();
+
+							return slowShutterStatus.orElse(null);
+						}
+						case SHUTTER_DIRECT:
```

Comment:
no need to _break_ here? It will fall down to next available case

Reply:
all of them share the same logic, so I put all of them into 1 case.